packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/BaseCompare.js

```diff
@@ -1,27 +1,131 @@
 /* eslint-disable class-methods-use-this, no-unused-vars */
+import path from 'path';
+
+const TEST_ID_REGEX = /\[([^)]+)\]/;
 
 export default class BaseCompare {
+  constructor(options) {
+    const {
+      baseScreenshotDir,
+      locale,
+      theme,
+    } = options;
+
+    // screenshot naming config
+    this.baseScreenshotDir = baseScreenshotDir || process.cwd();
+    this.locale = locale || 'en';
+    this.theme = theme || 'terra-default-theme';
+  }
+
   /**
-   * You can do here your image comparison magic.
+   * The subclass extending the BaseCompare class should implement this method to perform
+   * the screenshot processing appropriate for the subclass' comparison method.
+   *
+   * @param {Object} context - Information provided to process the screenshot.
+   * @param {Object} context.browser - Contains the browser's name, version, userAgent.
+   * @param {Object} context.suite - The test suite that is running.
+   * @param {Object} context.test - The test that is running.
+   * @param {Object} context.meta - Contains the element, exclude, hide, remove, viewport as meta data to use.
+   * @param {*} base64Screenshot - The screenshot captured by the selenium command to process.
    */
-  async processScreenshot(context, base64Screenshot) {
+  async processScreenshot(_context, _base64Screenshot) {
     return Promise.resolve();
   }
 
+  /**
+   * Creates the sanitized test name for the screenshot.
+   *
+   * @param {String} fullName - The test name.
+   * @returns {String} - The test name.
+   */
+  createTestName(fullName) {
+    const matches = TEST_ID_REGEX.exec(fullName);
+
+    // If test ID is provided, use the ID for a shorter test name, otherwise use the full name.
+    let name = matches ? matches[1] : fullName.trim();
+
+    // Remove white space.
```

Comment:
nit: This comment is a little misleading. It replaces white space with underscores